lib/SMTLIB2/Solver.cpp

```diff
@@ -126,12 +140,47 @@ struct SMTLIBParser {
 
     if (!consumeExpected(')', ErrStr))
       return APInt();
+
+    return APInt(Width, StringRef(NumBegin, NumEnd - NumBegin), 10);
+  }
+
+  APInt parseHexBitVector(std::string &ErrStr) {
+    ErrStr.clear();
+    const char *NumBegin = Begin;
+    while (Begin != End && ((*Begin >= '0' && *Begin <= '9') ||
+           (*Begin >= 'a' && *Begin <= 'f') ||
+           (*Begin >= 'A' && *Begin <= 'F')))
+      ++Begin;
+    const char *NumEnd = Begin;
+    unsigned Width = NumEnd - NumBegin;
+
+    return APInt(Width*4, StringRef(NumBegin, Width), 16);
+  }
+
+  APInt parseModel(std::string &ErrStr) {
+    APInt Res;
+
+    if (!consumeExpected('(', ErrStr))
+      return Res;
+    if (!consumeExpected('(', ErrStr))
+      return Res;
+    if (!consumeSExpr(ErrStr))
+      if (!consumeBvName(ErrStr))
+        return Res;
+    if (consumeExpected('#', ErrStr)) {
+      if (consumeExpected('x', ErrStr))
+        Res = parseHexBitVector(ErrStr);
+      else
+        Res = parseBinaryBitVector(ErrStr);
```

Comment:
@jeroenk are you OK with assuming 'b' was consumed here? Otherwise, it requires bigger changes in the whole parsing code.